### ISSUE TITLE
Possible to have more than 26 different types of piece in a variant?

### ISSUE THREAD

**Issue by TomFryers** [ianfab/Fairy-Stockfish#216]
One for each letter of the alphabet defined in the ini file. I suppose this a pretty niche use-case, but is this currently possible, and if not is it feasible?

**Comment by ianfab** [comment]
Currently this is not possible, at least not unless some of the pieces only exist as promoted forms of pieces (like in shogi). However, since there currently are not that much more than 26 piece types anyway, I agree that this is a rather niche use-case. This might get more relevant when https://github.com/ianfab/Fairy-Stockfish/issues/58 is implemented. Is the question hypothetical, or do you have a concrete variant in mind?

**Comment by TomFryers** [comment]
The original reason this became a problem is that [a simple GUI I’ve been making](https://gitlab.com/Tom_Fryers/fairyboard) defines its own variant with 26 pieces. This is so that when a piece is placed the variant doesn’t need redefining, as doing that causes a small delay. I can work-around this by defining a new variant whenever a new piece type is placed, and other changes I’ve made mean it’s not now as big of a delay as it was when originally suggested it.

However, I realised that it would also be useful for dai shogi (and the even bigger variants), which has 29 different unpromoted pieces. The board for dai shogi is 15×15, so this would only really be useful after #38 is addressed. It would also depend on implementing the lion, which might be tricky.

**Comment by musketeerchess** [comment]
Hi Tom, Fabian
Does your GUI Handles Musketeer Stockfish also?

I wish all of you a very nice and fruitful coming new year.

Zied

Le mar. 22 déc. 2020 à 13:01, Tom Fryers <notifications@github.com> a
écrit :

**Comment by atollk** [comment]
Hey. Since the question above was whether the proposal is hypothetical or an actual requirement: for me, I would like to have asymmetrical "armies" of 16 custom pieces, i.e. up to 32 pieces in total.

**Comment by HGMuller** [comment]
WinBoard/XBoard recognizes 'dressed letters' as piece ID to get around the limitations of the alphabet: effectively new letters are created by suffixing them with quotes or exclamation point. So whenever something like L', L" or L! is encountered when reading piece ID, these are considered different letters from a simple L (and from each other). This quadruples the number of possible piece IDs.
